src/components/List.jsx

```diff
@@ -0,0 +1,129 @@
+import styled from "styled-components";
+import TodoItem from "./TodoItem";
+import { useState, useMemo } from "react";
+
+const List = ({ todos, onUpdate, setTodos, removeTodo }) => {
+  const [search, setSearch] = useState("");
+
+  const onChangeSearch = (e) => {
+    setSearch(e.target.value);
+  };
+
+  const getFilteredDate = () => {
+    if (search === "") {
+      return todos;
+    }
+    return todos.filter((todos) => {
+      return todos.content.toLowerCase().includes(search.toLowerCase());
+    });
+  };
+
+  const filteredTodos = getFilteredDate();
+
+  const { doneCount, notDoneCount } = useMemo(() => {
```

Comment:
doneCount와 notDoneCount 계산 부분을 useMemo를 활용해서 효율적으로 코드를 작성해주신 것 같아요! todos 상태가 변경될 때만 계산이 일어나니까 불필요한 연산이 많이 줄어들 것 같습니다. 다만 지금 filteredTodos는 getFilteredDate() 함수에서 계속 계산되고 있는 것 같아요. 이 부분도 useMemo로 캐싱해서 todos나 search가 바뀔 때만 계산하도록 수정하면 더욱 좋을 것 같습니다..!!

```suggestion
  const filteredTodos = useMemo(() => getFilteredDate(), [todos, search]);

  const { doneCount, notDoneCount } = useMemo(() => {
```